modules/nextflow/src/main/groovy/nextflow/script/ScriptMeta.groovy

```diff
@@ -150,20 +150,11 @@ class ScriptMeta {
     }
 
     void checkComponentName(ComponentDef component, String name) {
-        if( component !instanceof ProcessDef && component !instanceof FunctionDef ) {
+        if( component !instanceof WorkflowDef && component !instanceof ProcessDef && component !instanceof FunctionDef ) {
             return
         }
-        if (functionsCount.get(name)) {
-            final msg = "A function with name '$name' is defined more than once in module script: $scriptPath -- Make sure to not define the same function as process"
-            if (NF.isStrictMode())
-                throw new DuplicateModuleFunctionException(msg)
-            log.warn(msg)
-        }
-        if (imports.get(name)) {
-            final msg = "A process with name '$name' is defined more than once in module script: $scriptPath -- Make sure to not define the same function as process"
-            if (NF.isStrictMode())
-                throw new DuplicateModuleFunctionException(msg)
-            log.warn(msg)
+        if( functionsCount.get(name) || imports.get(name) ) {
+            throw new DuplicateModuleFunctionException("A component (function, process, or workflow) named '$name' is defined more than once in module script: $scriptPath -- Make sure to not define multiple components with the same name")
```

Comment:
Was not better keeping separate branches to have a more precise error message?

Reply:
Well, they weren't precise to begin with. The branch tells you whether you are overwriting a function or an imported component (which could be function, process, or workflow). So if you wanted to be more precise, you would need to inspect the specific type of both `imports.get(name)` and `component` and produce an error message accordingly.

I thought about doing that, but seemed like overkill at the time

Reply:
I could e.g. implement a `getTypename()` method on each ComponentDef subclass and generate an error message from that